tests/integration/tier1/convert-offline-systems/prepare_system.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+
+# Replace urls in rhsm.conf file to the dogfood satellite server
+def replace_urls_rhsm(satellite_url):
+    with open("/etc/rhsm/rhsm.conf", "r+") as f:
+        file = f.read()
+        # Replacing the urls
+        file = re.sub("hostname = .*", "hostname = {}".format(satellite_url), file)
+        file = re.sub("baseurl = .*", "baseurl = https://{}/pulp/repos".format(satellite_url), file)
+
+        # Setting the position to the top of the page to insert data
+        f.seek(0)
+        f.write(file)
+        f.truncate()
+
+
+# Configure and limit connection to the satellite server only
+def configure_connection(satellite_ip):
+    with open("/etc/dnsmasq.conf", "a") as f:
+        # Satellite url
+        f.write("address=/dogfood.sat.engineering.redhat.com/{}".format(satellite_ip))
+
+        # Everything else is resolved to localhost
+        f.write("address=/#/127.0.0.1")
+
+    with open("/etc/resolv.conf", "w") as f:
+        f.write("nameserver 127.0.0.1")
+
+
+def test_prepare_system(shell):
+    assert shell("yum install dnsmasq bind-utils wget -y").returncode == 0
+
+    satellite_url = "dogfood.sat.engineering.redhat.com"
+
+    # Install katello package
+    pkg_url = "https://dogfood.sat.engineering.redhat.com/pub/katello-ca-consumer-latest.noarch.rpm"
+    pkg_dst = "/usr/share/convert2rhel/subscription-manager/katello-ca-consumer-latest.noarch.rpm"
+    assert shell("wget --no-check-certificate --output-document {} {}".format(pkg_dst, pkg_url)).returncode == 0
+    assert shell("rpm -i {}".format(pkg_dst)).returncode == 0
+
+    replace_urls_rhsm(satellite_url)
+
+    satellite_ip = shell("dig +short {} | tail -1".format(satellite_url)).output
```

Comment:
can this be read by python?

Reply:
You mean getting the IP address?

Reply:
yes

Reply:
Ok, now it's done in python using the socket library.
https://github.com/oamg/convert2rhel/pull/310/files#diff-22d0a5a44d1b15bb0da0c4a1baec18b61cf6328540432e780c0085df7b6042d4R45